testutil/promrated/prometheus.go

```diff
@@ -41,6 +81,73 @@ func serveMonitoring(addr string) error {
 	return errors.Wrap(server.ListenAndServe(), "failed to serve prometheus metrics")
 }
 
+func getPubkeyToClusterInfo(ctx context.Context, promAuth string) (map[string]prometheus.Labels, error) {
+	res, err := getPromClusters(ctx, promAuth)
+	if err != nil {
+		return nil, err
+	}
+
+	if res.StatusCode == 200 {
+		result, err := readPromJSON(ctx, res)
+		if err != nil {
+			return nil, err
+		}
+
+		keyToClusterLabels := make(map[string]prometheus.Labels)
+
+		for _, cluster := range result.Data.Result {
+			if cluster.Metric.ClusterInfo != nil && cluster.Metric.ClusterName != "" {
+				keyToClusterLabels[cluster.Metric.PubKey] = map[string]string{
+					"cluster_name":    cluster.Metric.ClusterName,
+					"cluster_hash":    cluster.Metric.ClusterHash,
+					"cluster_network": cluster.Metric.ClusterNetwork,
+					"cluster_peer":    cluster.Metric.ClusterPeer,
+					"pubkey":          cluster.Metric.PubKey,
+				}
+			}
+		}
+
+		return keyToClusterLabels, nil
+	}
+
+	return nil, errors.New("error processing prom metrics", z.Str("url", res.Request.RequestURI))
+}
+
+func getPromClusters(ctx context.Context, promAuth string) (*http.Response, error) {
```

Comment:
returning a http response isn't a good abstraction, rather colocate http wrangling stuff and return proper business logic types